File: cpp/src/benchmark/operation/RdmaReadLatencyOperation.cpp
#include "RdmaReadLatencyOperation.h"

namespace Observatory {

RdmaReadLatencyOperation::RdmaReadLatencyOperation(Benchmark *benchmark, Benchmark::Mode mode, uint32_t operationCount, uint32_t operationSize) :
        RdmaLatencyOperation(benchmark, mode, operationCount, operationSize, Benchmark::RdmaMode::WRITE) {}

const char* RdmaReadLatencyOperation::getClassName() const {
    return "RdmaReadLatencyOperation";
}

const char* RdmaReadLatencyOperation::getOutputFilename() const {
    return "RDMA Read Latency";
}

}

File: java/benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/RdmaReadLatencyOperation.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.Connection.Mode;
import de.hhu.bsinfo.observatory.benchmark.Connection.RdmaMode;

public class RdmaReadLatencyOperation extends RdmaLatencyOperation {

    RdmaReadLatencyOperation(Connection connection, Mode mode, int operationCount, int operationSize) {
        super(connection, mode, operationCount, operationSize, RdmaMode.READ);
    }

    @Override
    String getOutputFilename() {
        return "RDMA Read Latency";
    }
}
